giraffe/src/utils/formatters.ts

```diff
@@ -153,7 +161,11 @@ export const timeFormatter = ({
         }
         return String(Number(lhour))
       }
-      return lhour
+      if (is24hourLocale || UTC_TIME_ZONE.test(timeZone) || hour12 === false) {
+        const numericalHour = Number(lhour) % 24
+        return numericalHour < 10 ? `0${numericalHour}` : `${numericalHour}`
```

Comment:
This seems really similar to line 150, but it relies upon a different implementation. I’m curious if we should make them uniform with respect to how they determine if it’s a single digit number or a two digit number

Reply:
>I’m curious if we should make them uniform with respect to how they determine if it’s a single digit number or a two digit number

@Asalem1 i think we should not. these are two separate cases and the code _happens_ be similar _now_.  we should let them be separate and see if more instances like this crop up, and if they do, we should make a decision then. two data points that happen to look similar is not enough signal to make a good abstraction.

A bad abstraction is way more deleterious than duplicated code, because duplicated code is just code. An abstraction is a concept in the system that takes mental work to understand and change.

Reply:
They are different. In line 150, the format is 'hh' lowercase, which does not display leading zeroes. 'hh' means 1 through 12.

Here, the format is 'HH' uppercase, which can display 0 through 23. We put a leading zero here as a hint to the user that the format goes from 0 to 23.